### PR TITLE
Ecs for editor gizmo

### PR DESCRIPTION
**Pull request by NPatel10**
## Summary

- Make the Gizmo in the editor to work on the ECS.
- Converted Input management to be compatible with the ECS and Engine

## Checklist
- [x] Pre-push checks pass `npm run check`
  - [x] Linter passing via `npm run lint`
  - [x] Unit & Integration tests passing via `npm run test:packages`
  - [x] Docker build process passing via `npm run build-client`
- [x] Changes have been manually QA'd
- [ ] Changes reviewed by at least 2 approved reviewers

## References

#3139 - Editor uses ECS for gizmos

## Reviewers

@HexaField @speigg

### CODE REVIEW

File: packages/editor/src/functions/createCameraEntity.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import { Engine } from '@xrengine/engine/src/ecs/classes/Engine'
+import { Entity } from '@xrengine/engine/src/ecs/classes/Entity'
+import { addComponent } from '@xrengine/engine/src/ecs/functions/ComponentFunctions'
+import { createEntity } from '@xrengine/engine/src/ecs/functions/EntityFunctions'
+import { Object3DComponent } from '@xrengine/engine/src/scene/components/Object3DComponent'
+import { Vector3 } from 'ethereal'
```
Comment (HexaField) [inline packages/editor/src/functions/createCameraEntity.ts:6]: incorrect import

**Review by HexaField — COMMENTED** [review]
Looking fantastic so far!

File: packages/editor/src/functions/createCameraEntity.ts (hunk 1 above)
Comment (NPatel10) [reply packages/editor/src/functions/createCameraEntity.ts]: Changed to `three`

**Review by HexaField — APPROVED** [review]
Great work.

**Pull request merged**